domain/sequence.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package domain
```

Comment:
Probably should be in a sequence package.

We generally don't have any "state" based thing in the domain's root.

Reply:
ok